rclpy/src/rclpy/node.cpp

```diff
@@ -62,4 +69,192 @@ get_node_namespace(py::capsule pynode)
+  return pyparameter_cls(pyname, type, value);
+}
+
+/// Populate a dict with parameters by node name
+/**
+ * \param[in] params the parameters for multiple nodes
+ * \param[in] pyparameter_cls the Parameter class
+ * \param[in] pyparameter_type_cls the Parameter.Type class
+ * \param[out] pynode_params a dictionary to populate with node names and parameters
+ */
+void
+_populate_node_parameters_from_rcl_params(
+  const rcl_params_t * params, py::object pyparameter_cls,
+  py::object pyparameter_type_cls, py::dict pynode_params)
+{
+  for (size_t i = 0; i < params->num_nodes; ++i) {
+    std::string node_name{params->node_names[i]};
+    if (node_name.empty()) {
+      throw std::runtime_error("expected node name to have at least one character");
+    }
+
+    // Make sure all node names start with '/'
+    if ('/' != node_name.front()) {
+      node_name.insert(node_name.begin(), '/');
+    }
+    auto pynode_name = py::str(node_name);
+
+    // make a dictionary for the parameters belonging to this specific node name
+    if (!pynode_params.contains(pynode_name)) {
+      pynode_params[pynode_name] = py::dict();
+    }
+    auto parameter_dict = pynode_params[pynode_name].cast<py::dict>();
+
+    rcl_node_params_t node_params = params->params[i];
+    for (size_t ii = 0; ii < node_params.num_params; ++ii) {
+      auto pyparam_name = py::str(node_params.parameter_names[ii]);
+
+      parameter_dict[pyparam_name] = _parameter_from_rcl_variant(
+        pyparam_name, &node_params.parameter_values[ii], pyparameter_cls, pyparameter_type_cls);
+    }
+  }
+}
+
+/// Populate a Python dict with node parameters parsed from CLI arguments
+/**
+ * \param[in] args CLI arguments to parse for parameters
+ * \param[in] pyparameter_cls the Parameter class
+ * \param[in] pyparameter_type_cls the Parameter.Type class
+ * \param[out] params_by_node_name A Python dict object to place parsed parameters into.
+ */
+void
+_parse_param_overrides(
+  const rcl_arguments_t * args, py::object pyparameter_cls,
+  py::object pyparameter_type_cls, py::dict pyparams_by_node_name)
+{
+  rcl_params_t * params = nullptr;
+  if (RCL_RET_OK != rcl_arguments_get_param_overrides(args, &params)) {
+    throw RCLError("failed to get parameter overrrides");
+  }
+  if (params) {
+    RCPPUTILS_SCOPE_EXIT({rcl_yaml_node_struct_fini(params);});
+    _populate_node_parameters_from_rcl_params(
+      params, pyparameter_cls, pyparameter_type_cls, pyparams_by_node_name);
+  }
+}
+
+/// Get a list of parameters for the current node from rcl_yaml_param_parser
+/**
+ * On failure, an exception is raised and NULL is returned if:
+ *
+ * Raises ValueError if the argument is not a node handle.
+ * Raises RCLError if the parameters file fails to parse
+ *
+ * \param[in] pyparameter_cls The rclpy.parameter.Parameter class object.
+ * \param[in] node_capsule Capsule pointing to the node handle
+ * \return A dict mapping parameter names to rclpy.parameter.Parameter (may be empty).
+ */
```

Comment:
@sloretz nit: isn't this docblock duplicated in the header file?

Reply:
Removed duplicate docblock in 33c40a34f62ff30c6490a03a31222d40250e0fc6